crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -626,25 +626,30 @@ impl<'this, TX: DbTxMut<'this> + DbTx<'this>> DatabaseProvider<'this, TX> {
         &self,
         keys: impl IntoIterator<Item = T::Key>,
     ) -> std::result::Result<usize, DatabaseError> {
-        self.prune_table_with_iterator_in_batches::<T>(keys, usize::MAX, |_| {})
+        self.prune_table_with_iterator_in_batches::<T>(keys, usize::MAX, |_| {}, |_| false)
     }
 
     /// Prune the table for the specified pre-sorted key iterator, calling `chunk_callback` after
     /// every `batch_size` pruned rows.
     ///
+    /// `keep_filter` can be used to skip pruning certain elements.
```

Comment:
let's either make it a `skip_filter` or change the comment, because it's a bit ambiguous imo

Reply:
true